mllib/src/main/scala/org/apache/spark/mllib/feature/ChiSqSelector.scala

```diff
@@ -197,3 +197,28 @@ class ChiSqSelector @Since("1.3.0") (
     new ChiSqSelectorModel(indices)
   }
 }
+
+/**
+ * Creates a ChiSquared feature selector by False Positive Rate (FPR) test.
+ * @param alpha the highest p-value for features to be kept
+ */
+@Since("2.1.0")
+class ChiSqSelectorByFpr @Since("2.1.0") (
+  @Since("2.1.0") val alpha: Double) extends Serializable {
+
+  /**
+   * Returns a ChiSquared feature selector by FPR.
+   *
+   * @param data an `RDD[LabeledPoint]` containing the labeled dataset with categorical features.
+   *             Real-valued features will be treated as categorical for each distinct value.
+   *             Apply feature discretizer before using this function.
+   */
+  @Since("2.1.0")
+  def fit(data: RDD[LabeledPoint]): ChiSqSelectorModel = {
+    val indices = Statistics.chiSqTest(data)
+      .zipWithIndex.filter { case (res, _) => res.pValue < alpha }
```

Comment:
Yes that also makes sense and works. One down side is that if I for some reason wanted to do different types of chi-squared selection I'd have to make two (or later, more) models, when conceptually there's only one chi-squared test underlying all of them. That seemed a bit funny, but not crazy.

I suppose I'm thinking that a next logical step is to simply also expose the p-values to the caller as well. Right now there's no way to tell just how important the features are; you only know which were selected. And then, it wouldn't make sense to have different selectors because they'd all be doing the same work, so the logic would belong in the model. Is that compelling?